subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/impl/HierarchicalFileWatcherUpdater.java

```diff
@@ -177,4 +201,9 @@ private void updateWatchedHierarchies(Set<Path> newHierarchiesToWatch) {
             .forEach(hierarchies::add);
         return hierarchies;
     }
+
+    public enum ReportedFileEventPath {
+        CANONICAL_PATH,
+        ABSOLUTE_PATH
+    }
```

Comment:
Instead of an enum and some `if`s we could inject an `EventLocationValidator` that could be a no-op for Windows and a validator for macOS.

Reply:
I am using a strategy now.